### PR TITLE
Fix flaky test in service_domain_db_SUITE

### PR DESCRIPTION
**Pull request by arcusfelis**
Fix a race condition in `service_domain_db_suite.db_keeps_syncing_after_cluster_join` testcase.

Our tests assume that sending message from mim() to mim2() node is instant.
But it is not true, so check_for_updates message is received after sync_local call in tests.
I.e. the following scenario fails:

1. Test node asks `mim2()` to insert a domain.
2. `mim2()` sends `check_for_updates` to `mim()` node.
3. Test node calls `sync_local` on mim() node.
4. `mim()` calls sync_local and replies to the test node.
5. `mim()` finally receives `check_for_updates` from `mim2()`, send in step 2.

This PR addresses https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4066/183250/pgsql_cets.25.3-amd64/big/ct_run.test%4048ff31232359.2023-08-01_06.11.42/big_tests.tests.service_domain_db_SUITE.logs/run.2023-08-01_06.28.17/service_domain_db_suite.db_keeps_syncing_after_cluster_join.html

```erlang
test_server:ts_tc failed on line 1782
Reason: {test_case_failed,{[<<"example1.com">>,<<"example2.com">>,<<"example3.com">>],
                   [<<"example1.com">>,<<"example2.com">>,<<"example3.com">>,
                    <<"example4.com">>]}}
```

If we add some debug logging we can see that `ping` sometimes arrives earlier than `check_for_updates`:

```erlang
when=2023-08-01T11:31:32.523697+00:00 level=error 
what=ping pid=<0.42026.0> at=service_domain_db:handle_call/3:107 
(Ping from sync_local())

when=2023-08-01T11:31:32.523827+00:00 level=error what=check_for_updates pid=<0.42026.0> at=service_domain_db:handle_check_for_updates/3:143 
doms="[{<<\"example1.com\">>,<<\"dummy auth\">>}]" info_from_node=mongooseim2@localhost info_msg_host_type="dummy auth" info_msg_domain=example2.com 
(check_for_updates from insert_domain call)
```

Proposed changes include:
*  Actually check that `check_for_updates` is received by the local server, even if it is send by a remote server.

It is easy to reproduce the failure with `repeat_until_any_fail` for this testcase (and after the fix I cannot trigger the error anymore).

How the flaky test looks like:

```erlang
db_keeps_syncing_after_cluster_join(Config) ->
    HostType = dummy_auth_host_type(),
    %% GIVING mim1 and mim2 are not clustered.
    %% Ask mim1 to join mim2's cluster
    %% (and mongooseim application gets restarted on mim1)
    leave_cluster(Config),
    service_enabled(mim()),
    {ok, _} = insert_domain(mim(), <<"example1.com">>, HostType),
    {ok, _} = insert_domain(mim2(), <<"example2.com">>, HostType),
    sync(),
    %% Nodes don't have to be clustered to sync the domains.
    %% XXX Could fail here (flaky!)
    assert_domains_are_equal(HostType),
    %% WHEN Adding mim1 joins into a cluster
    %% (and mongooseim application gets restarted on mim1)
    join_cluster(Config),
    service_enabled(mim()),
    %% THEN Sync is successful
    {ok, _} = insert_domain(mim(), <<"example3.com">>, HostType),
    {ok, _} = insert_domain(mim2(), <<"example4.com">>, HostType),
    sync(),
    %% XXX Could fail here too (flaky!)
    assert_domains_are_equal(HostType).
```

So, there are two places `assert_domains_are_equal` could fail before the fix.

It does not change any server logic, it is only fixes the test (because the correct state is achieved eventually, the test is just checking the condition too fast) .

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/183471) / small_tests / 139130e5d30144129745217b0826f4fb24a4fe67
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4072/183471/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4072/183471/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-08-01_12.30.39/lib.mongooseim.logs/run.2023-08-01_12.30.40/suite.log.html)

File: src/domain/service_domain_db.erl [hunk 1]
```diff
@@ -73,18 +73,34 @@ start_link() ->
...
+%% even if they are sent by other nodes in the cluster.
+%% We have to do an RPC to ensure there is nothing pending in the distributed communication buffers.
+%% Used in tests.
 sync_local() ->
-    gen_server:call(?MODULE, ping).
+    LocalPid = whereis(?MODULE),
+    true = is_pid(LocalPid),
+    Nodes = [node(Pid) || Pid <- all_members()],
+    %% Ping from all nodes in the cluster
+    [pong = rpc:call(Node, ?MODULE, ping, [LocalPid]) || Node <- Nodes],
```
Comment (NelsonVides) [inline src/domain/service_domain_db.erl:98]: Why not `erpc:multical/4` instead maybe? Just a suggestion, otherwise I can merge as is.

**Review by NelsonVides — APPROVED** [review]

**Comment by NelsonVides** [comment]
Btw, super good PR description, love it 😄

File: src/domain/service_domain_db.erl (hunk 1 above)
Comment (arcusfelis) [reply src/domain/service_domain_db.erl:98]: just to keep it simple. 
It is basically 2 nodes in the tests anyway.
And this function is used only in tests :D

**Pull request merged**